src/views/connection.vue

```diff
@@ -9,41 +9,136 @@ export default {
     }
   },
   computed: {
-    ...mapState(['collectionName', 'databaseName', 'records', 'totalNumberOfResults'])
+    ...mapState([
+      'openedCollections',
+      'collectionName',
+      'databaseName',
+      'activeRecords',
+      'collectionRecords',
+      'totalNumberOfResults'
+    ])
+  },
+  watch: {
+    collectionName: function(newValue) {
+      if (this.collectionRecords[newValue]) {
+        this.queryJson = this.collectionRecords[newValue].query
+      } else {
+        this.queryJson = '{}'
+      }
+    }
```

Comment:
we should find a way to remove this watch statement. this.queryJson should always reflect the active collection's query in the store.